Source/ACE.Api.Host/Program.cs

```diff
@@ -12,15 +15,127 @@ public static void Main(string[] args)
+                try
+                {
+                    var server = WebApp.Start<Startup>(url);
+
+                    success = true;
+                }
+                catch (System.Reflection.TargetInvocationException ex)
+                {
+                    Console.WriteLine($"Failed to start Api.Host due to the following error:");
+                    Console.WriteLine($"{ex.ToString()}");
+                    // Console.ReadLine(); // need a readline or the error flashes without being seen
+                    Console.WriteLine($"attempting to fix... expect a UAC prompt to follow, click yes or this fix will fail.");
+                    string fixedurl = url;
+                    if (!fixedurl.EndsWith("/"))
+                        fixedurl += "/";
+                    string arguments = $"http add urlacl url={fixedurl} user={Environment.UserName}";
+                    ProcessStartInfo procStartInfo = new ProcessStartInfo("netsh", arguments);
+
+                    Process proc = new Process();
+
+                    procStartInfo.UseShellExecute = true;
+                    procStartInfo.CreateNoWindow = true;
+                    procStartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+                    procStartInfo.Verb = "runas";
+
+                    proc.StartInfo = procStartInfo;
+
+                    try
+                    {
+                        proc.Start();
+
+                        proc.WaitForExit();
+
+                        var server = WebApp.Start<Startup>(url); // try to start the api host a second time.
+
+                        success = true;
+                    }
+                    catch (Exception ex2)
+                    {
+                        Console.WriteLine($"Failed to fix TargetInvocationException with the following error:");
+                        Console.WriteLine($"{ex2.ToString()}");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Failed to start Api.Host due to the following error:");
+                    Console.WriteLine($"{ex.ToString()}");
+                    Console.ReadLine(); // need a readline or the error flashes without being seen
+                }
+
+                bool fwRuleFound = false;
+                string fwCommand = "add";
+                string fwRuleName = "ACEApiHost";
+                try
+                {
+                    INetFwPolicy2 firewallPolicy = (INetFwPolicy2)Activator.CreateInstance(Type.GetTypeFromProgID("HNetCfg.FwPolicy2"));
+                    INetFwRule firewallRule = firewallPolicy.Rules.OfType<INetFwRule>().Where(x => x.Name == fwRuleName).FirstOrDefault();
+
+                    if (firewallRule != null)
+                    {
+                        Console.WriteLine(firewallRule.Name + " rule has been found in Firewall Policy");
+                        fwRuleFound = true;
+                        fwCommand = "set";
```

Comment:
`fwCommand = "set";` will this be used if fwRuleFound is true?

Reply:
Yeah, I had anticipated that it would be needed to use "set" if we're modifying the found rule in the firewall policy but hadn't thought of a specific use case for it until...